ulib/fs/FStar_Monotonic_Heap.fs

```diff
@@ -0,0 +1,45 @@
+module FStar_Monotonic_Heap
+module Obj = FSharp.Compatibility.OCaml.Obj
+
+type heap = unit
+
+(* https://www.lexifi.com/blog/references-physical-equality *)
+type 'a ref = {
+  mutable contents: 'a;
+  id: int
+}
+
+type 'a mref = 'a ref
+            
+let emp =
+  ()
+
+(* Logical functions on heap *)
+(* TODO : complete the functions to have the same interface as in FStar.Heap.fsti *)
+
+let addr_of _ = Obj.magic ()
+let is_mm _ = Obj.magic ()
+
+(* let compare_addrs *)
+
+// HACK: We need to somehow make the implementation agree with the interface. Those types seem to be used only 
+//       in lemmas, so they shouldn't matter.
+type ('a, 'b, 'c, 'd) contains = 'a * 'b * 'c * 'd
+type ('a, 'b) addr_unused_in = 'a * 'b
+type ('a, 'b, 'c, 'd) unused_in = 'a * 'b * 'c * 'd
```

Comment:
@nikswamy So, this seems to work. But is this a right approach? 

My feeling is that this kind of dead code could be removed during extraction more aggresively. Or is it actually needed for something? Or perhaps was it hard to do and not really needed so far?